protocols/kad/src/behaviour/test.rs

```diff
@@ -239,13 +239,13 @@ fn query_iter() {
         // Ask the first peer in the list to search a random peer. The search should
         // propagate forwards through the list of peers.
         let search_target = PeerId::random();
-        let search_target_key = kbucket::Key::new(search_target.clone());
-        let qid = swarms[0].get_closest_peers(search_target.clone());
+        let search_target_key = kbucket::Key::from(search_target);
+        let qid = swarms[0].get_closest_peers(search_target.to_bytes());
```

Comment:
See the other comment about retaining the ability to pass `PeerId`s and `Multihash`es to this API.